### PR TITLE
Completed Task8

### PR DESCRIPTION
**Pull request by MarijaArnotiene87**

### CODE REVIEW

File: Homework8/Task1/MassiveElement.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    internal class MassiveElement
+    {
+        private static string? inputValue;
+
+        public static void ShowMassiveElement()
+        {
+            int[] massive = { 8, 7, 1, 4, 2 };
+            Console.WriteLine("Input index of element in massive:");
+            string? inputedValue = Console.ReadLine();
+            string? checkedValue = inputedValue.Equals(string.Empty) ? null : inputValue;
```
Comment (yauhenists) [inline Homework8/Task1/MassiveElement.cs]: You always get null here.
There is a typo in task, should be
string? checkedValue = inputedValue.Equals(string.Empty) ? null : inputedValue;

File: Homework8/Task1/MassiveElement.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            }
+            catch (ArgumentNullException ex)
+            {
+                Console.WriteLine($"Bad argument: {ex.Message}");
+            }
+            catch (IndexOutOfRangeException ex)
+            {
+                Console.WriteLine($"Out of range: {ex.Message}");
+            }
+            catch (Exception ex)
```
Comment (yauhenists) [inline Homework8/Task1/MassiveElement.cs:34]: Two more possible exceptions are missed

**Review by yauhenists — CHANGES_REQUESTED** [review]

File: Homework8/Task2/ExceptionGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿namespace Homework8.Task2
+{
+    public class ExceptionGenerator
+    {
+        public void GenerateException()
```
Comment (yauhenists) [inline Homework8/Task2/ExceptionGenerator.cs:5]: In the task said - method should throw an exception but not to handle

**Review by yauhenists — CHANGES_REQUESTED** [review]

File: Homework8/Task1/MassiveElement.cs (hunk 1 above)
Comment (MarijaArnotiene87) [reply Homework8/Task1/MassiveElement.cs]: Fixed

File: Homework8/Task2/ExceptionGenerator.cs (hunk 3 above)
Comment (MarijaArnotiene87) [reply Homework8/Task2/ExceptionGenerator.cs:5]: Fixed

File: Homework8/Task1/MassiveElement.cs (hunk 2 above)
Comment (MarijaArnotiene87) [reply Homework8/Task1/MassiveElement.cs:34]: Done

File: Homework8/Program.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Homework8
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            MassiveElement.ShowMassiveElement();
+            var exceptionGenerator = new ExceptionGenerator();
+            exceptionGenerator.GenerateException();
```
Comment (yauhenists) [inline Homework8/Program.cs]: Please handle the exception in more appropriate way - use all blocks for handling

**Review by yauhenists — CHANGES_REQUESTED** [review]

File: Homework8/Program.cs (hunk 4 above)
Comment (MarijaArnotiene87) [reply Homework8/Program.cs]: If I understood correctly - fixed.

Comment (yauhenists) [reply Homework8/Program.cs]: Missed task condition:
1. Use all blocks
2. Appropriate way - is catching specific exception

File: Homework8/Program.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace Homework8
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            MassiveElement.ShowMassiveElement();
+            var exceptionGenerator = new ExceptionGenerator();
+            try
```
Comment (yauhenists) [inline Homework8/Program.cs:12]: Not all task conditions are done

**Review by yauhenists — CHANGES_REQUESTED** [review]

File: Homework8/Program.cs (hunk 4 above)
Comment (MarijaArnotiene87) [reply Homework8/Program.cs]: Done

File: Homework8/Program.cs (hunk 5 above)
Comment (MarijaArnotiene87) [reply Homework8/Program.cs:12]: What task and which conditions are not done?
"Create a class that will contain the **method/property** that throws the given exception." Do I understand correctly, that either method or property should be created?

**Review by yauhenists — APPROVED** [review]

**Pull request merged**